content/tutorials/manage-the-rippled-server/troubleshooting/diagnosing-problems.md

```diff
@@ -0,0 +1,74 @@
+# Diagnosing Problems with rippled
+
+If you have having problems with `rippled`, the first step is to collect more information to accurately characterize the problem. From there, it can be easier to figure out a root cause and a fix.
+
+If your server does not start, see [rippled Server Won't Start](server-wont-start.html) for a list of some possible causes and fixes.
+
+The remainder of this document suggests steps for diagnosing the problem if your server starts successfully.
+
+## Get the server_info
+
+You can use the commandline to get server status information from the local `rippled` instance. For example:
+
+```
+rippled server_info
+```
+
+The response to this command has a lot of information, which is documented along with the [server_info method][].
+For troubleshooting purposes, the most important fields are (from most commonly used to least):
+
+- **`server_state`** - Most of the time, this field should show either `full` or `proposing` depending on whether it is [configured as a validator](run-rippled-as-a-validator.html). The value `connected` means that the server can communicate with the rest of the peer-to-peer network, but it does not yet have enough data to track progress of the shared ledger state. Normally, syncing to the state of the rest of the ledger takes about 5-15 minutes after starting.
+
+    - If your server remains in the `connected` state for hours, or returns to the `connected` state after being in the `full` or `proposing` states, that usually indicates that your server cannot keep up with the rest of the network. The most common bottlenecks are disk I/O and network bandwidth.
```

Comment:
If your server seems stuck in the `connected` state or `syncing` state, and you are using SSDs, then you should try switching to NuDB (if you are currently using RocksDB).

1. Stop the server (ctrl+c)
2. Edit `rippled.cfg`, updating the `[node_db]` stanza. Example:
```
[node_db]
type=nudb
path=/var/lib/rippled/db/nudb
open_files=2000
filter_bits=12
cache_mb=256
file_size_mb=8
file_size_mult=2
advisory_delete=0
```

To avoid conflicts, make sure the `path` is different from what you were using for RocksDB.

The `open_files`, `filter_bits`, `cache_mb`, `file_size_mb`, and `file_size_mult` settings do not apply to NuDB, but there's no harm in leaving them; and if they remain, it may be easier to switch back to RocksDB if needed.

Reply:
This is false. Either rocksdb or nudb should be fine on SSD. In fact, nobody should run rippled without server grade SSD's. RocksDB becomes unusable based on memory consumption after a large amount of history is gathered, however. Though I cannot remember how much offhand.